xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/SetFee.java

```diff
@@ -53,34 +57,108 @@ static ImmutableSetFee.Builder builder() {
    * The charge, in drops of XRP, for the reference transaction, as hex. (This is the transaction cost before scaling
    * for load.)
    *
-   * @return A hex {@link String} basefee value.
+   * @return A hex {@link String} baseFee value.
+   *
+   * @deprecated Prefer {@link #baseFeeDrops()} over this field because the XRPFees ammendment now serializes this
+   *   object's base fee into XRP drops.
    */
-  @JsonProperty("BaseFee")
-  String baseFee();
+  @Value.Default
+  @Deprecated
+  @JsonIgnore
+  default String baseFee() {
+    return baseFeeDrops().value().toString(16);
+  }
 
   /**
-   * The cost, in fee units, of the reference transaction.
+   * The charge, in drops of XRP, for the reference transaction (This is the transaction cost before scaling for load).
+   *
+   * <p>This field will also be populated with the {@code BaseFee} value from any {@link SetFee} transactions
+   * that occurred before the XRPFees amendment took effect.</p>
+   *
+   * @return An {@link XrpCurrencyAmount}.
+   */
+  @JsonProperty("BaseFeeDrops")
+  @JsonAlias("BaseFee")
+  @JsonDeserialize(using = BaseFeeDropsDeserializer.class)
+  XrpCurrencyAmount baseFeeDrops();
+
+  /**
+   * The cost, in fee units, of the reference transaction. This field can be {@code null} if this {@link SetFee}
+   * transaction was included in a ledger after the XRPFees amendment was enabled because this amendment removes the
+   * {@code ReferenceFeeUnits} field from the {@link SetFee} transaction.
    *
    * @return An {@link UnsignedInteger} cost of ref transaction.
+   *
+   * @deprecated Prefer {@link #maybeReferenceFeeUnits()} over this field.
+   */
+  @Deprecated
+  @Nullable
+  @Value.Default
+  @JsonIgnore
+  default UnsignedInteger referenceFeeUnits() {
```

Comment:
Remove deprecation, make a breaking change of just making this `Optional`, and get rid of `maybeReferenceFeeUnits`